report/3.Theory/Parser.tex

```diff
@@ -0,0 +1,112 @@
+\section{Parser}
+
+The second part of the compiler will parse the lexical tokens coming from the lexer. 
+This part of the compiler is called the parser. 
+The parser works by taking a sequence of lexical tokens, and generating a parse tree or transforming them into an abstract syntax tree. 
+
+\subsection{Output data types}
+
+The parser will output some data type, that represents the compiled program in a hierarchical structure. 
```

Comment:
fjern komma før "that"
eller ændr "that" til "which"

Engelsk kommatering er bare det bedste, ik? 😃 😃 😃